src/lib/gsheets/index.js

```diff
@@ -0,0 +1,286 @@
+  const promisesSheetId = server.env("GSHEETS_PROMISES_SHEET_ID");
+  const promisesCategoriesSheetId = server.env(
+    "GSHEETS_PROMISES_CATEGORIES_SHEET_ID"
+  );
+  const promisesEventsSheetId = server.env("GSHEETS_PROMISES_EVENTS_SHEET_ID");
+
+  const papaOptions = {
+    header: true,
+    transformHeader: (header) => camelCase(header.trim()),
+    transform: (value) => value?.trim(),
+  };
+
+  const papaPromise = (readableStream, options = papaOptions) => {
+    const parseStream = papa.parse(papa.NODE_STREAM_INPUT, options);
+    readableStream.pipe(parseStream);
+
+    return new Promise((resolve, reject) => {
+      const data = [];
+      parseStream.on("data", (chunk) => {
+        data.push(chunk);
+      });
+      parseStream.on("end", () => {
+        resolve(data);
+      });
+      parseStream.on("error", (error) => {
+        reject(error);
+      });
+    });
+  };
+
+  async function fetchSheet(sheetId, options) {
+    const response = await fetch(
+      `${SPREADSHEETS_URL}d/${spreadsheetId}/export?format=csv&gid=${sheetId}`
+    );
+    return papaPromise(response.body, options);
+  }
+
+  async function fetchSitesSheet() {
+    const options = {
+      ...papaOptions,
+      dynamicTyping: (header) =>
+        ["articles", "resources", "factchecks", "actnow"].includes(
+          header.toLowerCase()
+        ),
+    };
+    return fetchSheet(sitesSheetId, options);
+  }
+
+  async function fetchStatuses() {
+    const statusesSheet = await fetchSheet(statusesSheetId);
+    const { statusPalette } = theme;
+    const getPalette = (ordinal) =>
+      statusPalette[`status${ordinal}`] || statusPalette.status0;
+    return statusesSheet.map((row) => ({
+      ...row,
+      ...getPalette(row.ordinal),
+      // FIXME(kilemensi): Some components expect title instead of name
+      title: row.name,
+      slug: slugify(row.name),
+    }));
+  }
+
+  async function fetchCategories() {
+    const categoriesSheet = await fetchSheet(categoriesSheetId);
+    return categoriesSheet
+      .filter((row) => row.name)
+      .map((row) => ({
+        ...row,
+        // FIXME(kilemensi): Some components expect title instead of name
+        title: row.name,
+        slug: slugify(row.name),
+      }));
+  }
+
+  async function fetchSite() {
+    const statuses = await fetchStatuses();
+    const categories = await fetchCategories();
+    const entitiesSheet = await fetchSheet(entitiesSheetId);
+    const sitesSheet = await fetchSitesSheet();
+    const siteRow = server.slug
+      ? sitesSheet.find((row) => equalsIgnoreCase(server.slug, row.slug))
+      : sitesSheet[0];
+    const entity = entitiesSheet.find((row) =>
+      equalsIgnoreCase(row.name, siteRow.entity)
+    );
+    return { ...siteRow, categories, entity, statuses };
+  }
+
+  function formatSiteAsProjectMeta(site) {
+    const { entity } = site;
+    const description = `Campaign Promises made by ${entity.name}`;
```

Comment:
Will it always be a `Campaign Promise`? I think we can have `Promises made by ....`

Reply:
This whole method has been marked for [removal](https://github.com/CodeForAfrica/PromiseTracker/pull/349/files#diff-05619e230ef0ea4ff123db7e04ca5de1982a2f62380852aab5a14965e3d1e8a4R237-R246) @esirK